src/shared/components/oncoprint/DeltaUtils.ts

```diff
@@ -405,10 +415,30 @@ function transitionTracks(
             }
         }
     }
-
+    
+    // collect max trackId (last assigned one) for each treatment profile
+    // the last assigned track id is used to reference the current layout of the track group
+    // Note: the resolution of `trackIds for ruleset sharing` is different from
+    // the section above because different formatting is applied to each treatment profile (molecularProfileId)
+    trackIdForRuleSetSharing.treatment = _.chain(prevProps.heatmapTracks)
+                                    .filter((s:IHeatmapTrackSpec) => s.molecularAlterationType === AlterationTypeConstants.TREATMENT_RESPONSE)
+                                    .groupBy((track:IHeatmapTrackSpec) => track.molecularProfileId)
+                                    .mapValues( (o:IHeatmapTrackSpec[]) => _(o).map((d)=>d.key).map((key:string) => trackSpecKeyToTrackId[key]).max() )
```

Comment:
It's not technically part of the oncoprint API that more recent tracks are assigned higher ids

Reply:
It seems like you shouldn't bother setting these inside each transitionTrack as well, you should probably set them all at once at the end, right?

Reply:
Ok, I have refactored the evaluation of the trackID for the last added track. Instead of the max value I find the last added in the sequence:
```
    trackIdForRuleSetSharing.treatment = _.chain(prevProps.heatmapTracks)
                                    .filter((s:IHeatmapTrackSpec) => s.molecularAlterationType === AlterationTypeConstants.TREATMENT_RESPONSE)
                                    .groupBy((track:IHeatmapTrackSpec) => track.molecularProfileId)
                                    .mapValues( (o:IHeatmapTrackSpec[]) => _.last(o) )
                                    .mapValues( (o:IHeatmapTrackSpec) => trackSpecKeyToTrackId[_.get(o, 'key')] )
                                    .value();
```
This implementation relies on preservation of the order in the lodash `groupBy` function. This is a safe assumption because part of the function description:
```
Creates an object composed of keys generated from the results of running each element of collection thru iteratee. The order of grouped values is determined by the order they occur in collection. The corresponding value of each key is an array of elements responsible for generating the key. The iteratee is invoked with one argument: (value).
```

Reply:
I do not really understand this comment:

>It seems like you shouldn't bother calling shareRuleSet these inside each transitionTrack as well, you should probably set them all at once at the end, right?

Do you mean that I should suppress calling `shareRuleSet` for treatment tracks in the `transitionHeatmapTrack()` function and call id after the `// Transition heatmap tracks` block in `transitionTracks`?

Reply:
@pvannierop sorry for the confusion. this is what we talked about the other day and I agreed the way you're doing it makes sense